build.gradle

```diff
@@ -153,9 +156,68 @@ task installTestGems(dependsOn: downloadAndInstallJRuby, type: Exec) {
   environment "GEM_PATH", "1"
   standardOutput = new ExecLogOutputStream(System.out)
   errorOutput =  new ExecLogOutputStream(System.err)
-  commandLine jrubyBin, "${projectDir}/vendor/jruby/bin/rake", "test:install-core"
+  commandLine jrubyBin, rakeBin, "test:install-core"
+}
+
+task assembleTarDistribution(dependsOn: installTestGems, type: Exec) {
+  workingDir projectDir
+  inputs.files fileTree("${projectDir}/rakelib")
+  inputs.files fileTree("${projectDir}/logstash-core/lib")
+  inputs.files fileTree("${projectDir}/logstash-core/src")
+  outputs.files file("${buildDir}/logstash-${project.version}.tar.gz")
+  standardOutput = new ExecLogOutputStream(System.out)
+  errorOutput =  new ExecLogOutputStream(System.err)
+  commandLine jrubyBin, rakeBin, "artifact:tar"
+}
+
+task unpackTarDistribution(dependsOn: assembleTarDistribution, type: Copy) {
+  def tar = file("${buildDir}/logstash-${project.version}-SNAPSHOT.tar.gz")
+  inputs.files tar
+  outputs.files fileTree("${buildDir}/logstash-${project.version}-SNAPSHOT")
+  from tarTree(tar)
+  into {buildDir}
+}
+
+def bundleBin = "${projectDir}/vendor/bundle/jruby/2.3.0/bin/bundle"
+def gemPath = "${buildDir}/qa/integration/gems"
+
+task installIntegrationTestBundler(dependsOn: unpackTarDistribution, type: Exec) {
+  outputs.files fileTree("${gemPath}/gems/bundler-1.16.0")
+  environment "GEM_PATH", gemPath
+  environment "GEM_HOME", gemPath
+  standardOutput = new ExecLogOutputStream(System.out)
+  errorOutput =  new ExecLogOutputStream(System.err)
+  commandLine jrubyBin, "${projectDir}/vendor/jruby/bin/gem", "install", "bundler", "-v", "1.16.0"
+}
+
+task installIntegrationTestGems(dependsOn: installIntegrationTestBundler, type: Exec) {
+  workingDir "${projectDir}/qa/integration"
+  environment "GEM_PATH", gemPath
+  environment "GEM_HOME", gemPath
+  inputs.files file("${projectDir}/qa/integration/Gemfile")
+  inputs.files file("${projectDir}/qa/integration/integration_tests.gemspec")
+  outputs.files fileTree("${gemPath}/gems")
+  outputs.files file("${projectDir}/qa/integration/Gemfile.lock")
+  standardOutput = new ExecLogOutputStream(System.out)
+  errorOutput =  new ExecLogOutputStream(System.err)
+  commandLine jrubyBin, bundleBin, "update"
```

Comment:
did you mean to update (instead of install) ?

Reply:
@jakelandis hmm it makes no difference here actually since `Gemfile.lock` is supposed to be cleaned by Gradle anyways I guess. But I `install` is more aesthetically pleasing, I agree :)